### PR TITLE
Implement 'title' directive argument

### LINKED ISSUE #2: Possibility of not including the title
Thank you for this utility and the quick feedback, I've only one problem more to use this in in my project.

**Is your feature request related to a problem? Please describe.**
I've this RST structure:

```rst
***********************
Command line interfaces
***********************

foo
===

.. sphinx_argparse_cli::
  :module: foo.__main__
  :func: build_parser
  :prog: foo

bar
===

.. sphinx_argparse_cli::
  :module: bar.__main__
  :func: build_parser
  :prog: bar
```

That is rendered in HTML as:

```html
<h1>Command line interfaces</h1>

<h2>foo</h2>
<h3>foo - CLI Interface</h3>

... Rest of `h3` sections

<h2>bar</h2>
<h3>bar - CLI Interface</h3>

... Rest of `h3` sections
```

**Describe the solution you'd like**
I would like to remove the first `<h3>` rendered by `sphinx_argparse_cli`, so I can specify my own title in an `h2` tag. Something like:

```html
<h1>Command line interfaces</h1>

<h2>foo</h2>

... Rest of `h3` sections

<h2>bar</h2>

... Rest of `h3` sections
```

**Describe alternatives you've considered**
A boolean `include_title` directive, but I'm not sure if this is out of your expectations for this package or if you would like to solve it in another way :thinking:

### PR DESCRIPTION
**Pull request by mondeja**
*Issue number of the reported bug or feature request: Closes #2 

**Describe your changes**
Added `:title:` directive argument that, when provided, the `<prog> - CLI interface` title is replaced by the value, and if empty is not included.

**Testing performed**
Tested in two different new tests and manually in my own project.

**Additional context**
Let me know if you want another name for this argument.

### CODE REVIEW

**Review by gaborbernat — CHANGES_REQUESTED** [review]
I see you've missed my edit:

> PS. Though now that I'm thinking about it. Let's make this a title attribute of type text, which you can override. If not set use what we have now. If set and empty do not add the title. How about that?

Can we do this instead?

**Comment by mondeja** [comment]
Ok :+1: Let me update this pull

File: setup.cfg [hunk 1]
```diff
@@ -57,7 +57,7 @@ where = src
 
 [flake8]
 max-line-length = 120
-ignore = F401, H301, E203
+ignore = F401, H301, E203, SC200
```
Comment (gaborbernat) [inline setup.cfg:60]: Please don't do this 🙇🏻

File: tests/test_logic.py [hunk 2]
```diff
@@ -25,3 +25,12 @@ def test_prog_as_text(app: SphinxTestApp) -> None:
     app.build()
     outcome = (Path(app.outdir) / "index.txt").read_text()
     assert outcome == "magic - CLI interface\n*********************\n\n   magic\n"
+
+
+@pytest.mark.sphinx("text", testroot="noincludetitle")
+def test_noincludetitle_as_text(app: SphinxTestApp) -> None:
```
Comment (gaborbernat) [inline tests/test_logic.py:31]: I don't think you need here to define a new noincludetitle, you can just reuse basic 👍🏻

File: README.md [hunk 3]
```diff
@@ -31,16 +31,25 @@ extensions = ["sphinx_argparse_cli"]
...
   :module: a_project.cli
   :func: build_parser
   :prog: my-cli-program
 ```
+
+```rst
+.. sphinx_argparse_cli::
+  :module: a_project.cli
+  :func: build_parser
+  :title:
```
Comment (gaborbernat) [inline README.md:54]: I don't think we need this, not?

File: README.md [hunk 4]
```diff
@@ -31,16 +31,25 @@ extensions = ["sphinx_argparse_cli"]
...
 
-Within the reStructuredText files use the `sphinx_argparse_cli` directive that takes two arguments:
+Within the reStructuredText files use the `sphinx_argparse_cli` directive that takes, at least, two arguments:
 
 - the module path to where the parser is defined,
 - a no argument function within that module that once called returns the constructed
   [argparse](https://docs.python.org/3/library/argparse.html) parser
 - (optional) a program name that overwrites the autodiscovered running argument parser
+- (optional) a `:title:` argument which, when provided, overwrites the
+  `prog - CLI interface` title added by default and when empty, will not be included
```
Comment (gaborbernat) [inline README.md:41]: let's put prog in ``<prog>`` to show it's whatever your prog of the parser is

File: roots/test-title--empty/index.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.. sphinx_argparse_cli::
```
Comment (gaborbernat) [inline roots/test-title--empty/index.rst:1]: we can name this just ``test-title-empty``

File: roots/test-title/index.rst [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.. sphinx_argparse_cli::
```
Comment (gaborbernat) [inline roots/test-title/index.rst:1]: And this ``test-title-set``

**Review by gaborbernat — APPROVED** [review]

**Comment by gaborbernat** [comment]
https://github.com/gaborbernat/sphinx-argparse-cli/releases/tag/1.2.0

**Comment by gaborbernat** [comment]
Release in progress under https://github.com/gaborbernat/sphinx-argparse-cli/actions/runs/540540089

**Comment by mondeja** [comment]
> Release in progress under...

Perfect, thanks :clap:

**Pull request merged**